scripts/genproto.sh

```diff
@@ -15,41 +15,32 @@ if ! [[ $(protoc --version) =~ "3.5.1" ]]; then
 	exit 255
 fi
 
-# Exact version of plugins to build.
-PROTOC_GEN_GOGOFAST_SHA="971cbfd2e72b513a28c74af7462aee0800248d69"
-PROTOC_GEN_GRPC_ECOSYSTEM_SHA="e4b8a938efae14de11fd97311e873e989896348c"
-
 echo "installing plugins"
-go install "golang.org/x/tools/cmd/goimports"
-go get -d -u "github.com/gogo/protobuf/protoc-gen-gogo"
-pushd ${GOPATH}/src/github.com/gogo/protobuf
-    git reset --hard "${PROTOC_GEN_GOGOFAST_SHA}"
-    go install "github.com/gogo/protobuf/protoc-gen-gogofast"
-popd
+INSTALL_PKGS="golang.org/x/tools/cmd/goimports github.com/gogo/protobuf/protoc-gen-gogofast github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger"
+for pkg in ${INSTALL_PKGS}; do
+    GO111MODULE=on go install -mod=vendor "$pkg"
+done
 
-go get -d -u "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway"
-go get -d -u "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger"
-pushd ${GOPATH}/src/github.com/grpc-ecosystem/grpc-gateway
-    git reset --hard "${PROTOC_GEN_GRPC_ECOSYSTEM_SHA}"
-    go install "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway"
-    go install "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger"
-popd
+for pkg in github.com/gogo/protobuf github.com/grpc-ecosystem/grpc-gateway; do
+    GO111MODULE=on go get "$pkg"
```

Comment:
hmm. Those packages are listed in `go.mod` so I assume they are vendored. 

Since you go get without pinned version (no `@version` at the end), according to: https://github.com/golang/go/wiki/Modules#how-to-upgrade-and-downgrade-dependencies `go get` or install with modules will get or install `@latest` version and update `go.mod` with that.

I thought we want reproducible proto generation for all changes until manual update will be done. In this case don't we want to pin versions of those packages?

Reply:
Correct. For whatever reason, I assumed that `go get` would rely on `go.mod` information. `go mod download` should the trick instead.

Reply:
Nice!